_posts/2015-02-06-Utilising-API-Blueprint-in-API-Clients.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Utilising API Blueprint in API Clients
+excerpt: Imagine an API client when you specify which resource and action you want to perform from an API Blueprint.
+layout: post
+date: 2015-02-06 18:00:00 +0000
+author: kyle
+published: true
+comments: true
+---
+
+I’m excited to release a new way to interface to API Blueprints. A client-side framework in Swift that allows you to introspect and perform actions at run-time.
+
+Imagine an API client when you specify which resource and action you want to perform from an API Blueprint, it can use the already defined information from a blueprint to build the request without having to duplicate the information such as the HTTP method and URI from your blueprint. This allows you to start building smarter clients which can evolve at run-time based on the blueprint.
+
+The SDK is built on-top of a framework we’ve built called the [Representor](https://github.com/the-hypermedia-project/charter), which is designed to be a canonical representation of a hypermedia message in multiple languages. One of the key constraints to Representational State Transfer (REST) is Hypermedia as the Engine of State (HATEOS).  The message from the server itself should contain information about how to transition to other states along with links to other resources.
+
+The Representor is a structure providing this functionality along with the ability to deserialise Hypermedia formats such as [HAL](http://stateless.co/hal_specification.html) and [Siren](https://github.com/kevinswiber/siren). The Representer contains a couple of properties, one of them is it’s transitions to other states, along with how the current resource links to other resources.
+
+Since a blueprint is effectively a document which specifies all of the transitions (actions) in your API, we’ve built on-top of the Representor and added the ability to deserialise an API Blueprint into a collection of transitions.
+
+This allows us to decouple an API client from the server-side implementation, we no longer have to know about the URIs to each resource and the HTTP methods in our code. We can negotiate these at run-time, similar to how with HATEOS we follow links, we can follow links from an API Blueprint.
```

Comment:
After "implementation," I think that sentence should end.

Last sentence could read, since following links is HATEOAS (IMO):

We can negotiate these at run-time and follow links from an API Blueprint.